### PR TITLE
Benchmark API Improvements

### PR DESCRIPTION
**Pull request by gottesmm**
This PR updates parts of the benchmark harness to match the API guidelines closer. That includes using a String enum for BenchmarkCategories and renaming BenchmarkCategories to BenchmarkCategory (since it is an enum that represents a single benchmark category after all).

### CODE REVIEW

**Comment by gottesmm** [comment]
@atrick @shajrawi What do you guys think? While doing the initFunction change I noticed this.

**Comment by atrick** [comment]
👍 Obvious goodness. You also mentioned moving the runFunction to the end of the initializer.

File: benchmark/utils/DriverUtils.swift [hunk 1]
```diff
@@ -150,75 +152,16 @@ struct TestConfig {
...
-      if x.contains("algorithm") {
-        tags.insert(BenchmarkCategories.algorithm)
-      }
-      if x.contains("miniapplication") {
-        tags.insert(BenchmarkCategories.miniapplication)
-      }
-      if x.contains("regression") {
-        tags.insert(BenchmarkCategories.regression)
+      if x.isEmpty { return .fail("--tags requires a value") }
+      guard let cat = BenchmarkCategory(rawValue: x) else {
```
Comment (shajrawi) [inline benchmark/utils/DriverUtils.swift]: What about having multiple categories as an input to the driver? The current implementation allows having more than one category inside the set.

Comment (shajrawi) [reply benchmark/utils/DriverUtils.swift]: for example, I can pass `api,Array,regression` as tag's value something I can't do anymore unless I am missing something this late at night?

Comment (gottesmm) [reply benchmark/utils/DriverUtils.swift]: That doesn't work today.

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@swift-ci smoke test

File: benchmark/utils/DriverUtils.swift (hunk 1 above)
Comment (gottesmm) [reply benchmark/utils/DriverUtils.swift]: But you are right, we /should/ support multiple tags. I actually was surprised when I checked that we didn't. The way you can see is that the optional argument/etc is a `[String : String]`

Comment (shajrawi) [reply benchmark/utils/DriverUtils.swift]: That *DID* work before merging the commit.

Comment (gottesmm) [reply benchmark/utils/DriverUtils.swift]: I talked with @shajrawi offline. Turns out we were expecting people to pass tags using a comma delimited list. I will fix this.

**Pull request merged**